src/Examine.AzureDirectory/AzureReadOnlyLuceneDirectory.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Azure.Storage.Blobs;
+using Examine.LuceneEngine;
+using Examine.LuceneEngine.Directories;
+using Lucene.Net.Store;
+using Directory = Lucene.Net.Store.Directory;
+
+namespace Examine.AzureDirectory
+{
+    public class AzureReadOnlyLuceneDirectory : AzureLuceneDirectory
+    {
+        private readonly string _cacheDirectoryPath;
+        private readonly string _cacheDirectoryName;
+        private string OldIndexFolderName;
+
+        public AzureReadOnlyLuceneDirectory(
+            string storageAccount,
+            string containerName,
+            string cacheDirectoryPath,
+            string cacheDirectoryName,
+            bool compressBlobs = false,
+            string rootFolder = null) : base(storageAccount, containerName, null, compressBlobs, rootFolder, true)
+        {
+            _cacheDirectoryPath = cacheDirectoryPath;
+            _cacheDirectoryName = cacheDirectoryName;
+            if (CacheDirectory == null)
+            {
+                Trace.WriteLine("INFO CacheDirectory null. Creating or rebuilding cache");
+                CreateOrReadCache();
```

Comment:
Do we want to perform operations on construction? IMO constructors should be cheap, perhaps this should be done in a lazy way when the index needs to do something?

Reply:
@Shazwazza lazyloading of that could cause issues, it is why i decided to put that in constructor, as we can kill application which are not aware that is not lazy load ;/ but there we can discuss things